src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd.MM.yyyy");
+        Map<String, Integer> results = new HashMap<>();
+
+        for (String name : names) {
+            results.put(name, 0);
+        }
+
+        try {
+            Date startDate = simpleDateFormat.parse(dateFrom);
+            Date endDate = simpleDateFormat.parse(dateTo);
+
+            for (String row : data) {
+                String[] rowData = row.split(" ");
+
+                Date parsedDate = simpleDateFormat.parse(rowData[0]);
+                String name = rowData[1];
+                int hours = Integer.parseInt(rowData[2]);
+                int salary = Integer.parseInt(rowData[3]);
+
+                if (results.containsKey(name)
+                        && parsedDate.after(startDate)
+                        && (parsedDate.before(endDate) || parsedDate.equals(endDate))
+                ) {
+                    results.replace(name, results.get(name) + hours * salary);
+                }
+            }
+        } catch (ParseException e) {
```

Comment:
The task doesn't provide for a wrong date format in the input data. But even if you want to handle this case, you need to keep the smallest part of the code where this error could potentially occur in the try block